src/ui/server/controllers/insurance/your-buyer/trading-history/index.ts

```diff
@@ -262,22 +125,42 @@ export const post = async (req: Request, res: Response) => {
       return res.redirect(PROBLEM_WITH_SERVICE);
     }
 
+    const answer = payload[FIELD_ID];
+
+    const hasTradingHistory = answer === 'true';
+
     /**
      * If is a change route
      * redirect to CHECK_YOUR_ANSWERS
      */
-    if (isChange) {
+    if (isChangeRoute(req.originalUrl)) {
+      if (hasTradingHistory) {
+        return res.redirect(`${INSURANCE_ROOT}/${referenceNumber}${CURRENCY_OF_LATE_PAYMENTS_CHANGE}`);
+      }
+
       return res.redirect(`${INSURANCE_ROOT}/${referenceNumber}${CHECK_YOUR_ANSWERS}`);
     }
 
     /**
      * If is a check-and-change route
      * redirect to CHECK_AND_CHANGE_ROUTE
      */
-    if (isCheckAndChange) {
+    if (isCheckAndChangeRoute(req.originalUrl)) {
+      if (hasTradingHistory) {
+        return res.redirect(`${INSURANCE_ROOT}/${referenceNumber}${CURRENCY_OF_LATE_PAYMENTS_CHECK_AND_CHANGE}`);
+      }
+
       return res.redirect(`${INSURANCE_ROOT}/${referenceNumber}${CHECK_AND_CHANGE_ROUTE}`);
     }
 
+    /**
+     * if OUTSTANDING_PAYMENTS is true
+     * then should redirect to CURRENCY_OF_LATE_PAYMENTS
+     */
+    if (hasTradingHistory) {
+      return res.redirect(`${INSURANCE_ROOT}/${referenceNumber}${CURRENCY_OF_LATE_PAYMENTS}`);
+    }
```

Comment:
Can we review the documentation here - the last one mentions
`OUTSTANDING_PAYMENTS`, but it's for trading history

Can we In the previous conditions above, can we compare with this controller that mentions e.g "and if X is true": `export-contract/private-market/index.ts`